app/views/fragments/ABTest.scala

```diff
@@ -0,0 +1,19 @@
+package views.fragments
+
+import play.api.mvc.{Cookie, Request}
+
+import scala.util.Random
+
+object ABTest {
+  val TestIdCookieName = "gu.subscriptions.test.id"
+
+  def testIdFor[A](implicit request: Request[A]): Int = {
+
+    request.getQueryString("stripe").flatMap((str) => if (str == "checkout") {Some(2)} else {Some(1)}).
+      getOrElse(((request.cookies.get(TestIdCookieName) map (_.value.toInt)).
+        getOrElse(Random.nextInt(Int.MaxValue))))
```

Comment:
If you use 0 and 1 for on/off then you can do Random.nextInt(2) here and be guaranteed to get a number back that you can support.

def nextInt(n : Int) : Int

    Returns a pseudorandom, uniformly distributed int value between 0 (inclusive) and the specified value (exclusive), drawn from this random number generator's sequence.